src/searching/binary_search_recursive.rs

```diff
@@ -0,0 +1,226 @@
+use std::cmp::Ordering;
+/// ## Binary Search algorithm
+/// Iterating over the array by dividing half of the array and returns the index value of the target element in the array.
+/// you can pass both ascending and decending array respectively.
```

Comment:
Same as above, the implementation/documentation shouldn't care about this property of the array. We can assume it's sorted in non-decreasing order.